Meta/CMake/lagom_compile_options.cmake

```diff
@@ -10,9 +10,20 @@ add_compile_options(-fno-exceptions)
 add_compile_options(-fdiagnostics-color=always)
 add_compile_options(-g1)
 add_compile_options(-O2)
-if (NOT ENABLE_FUZZERS AND NOT APPLE)
+if (NOT ENABLE_FUZZERS AND NOT APPLE AND NOT WIN32)
     add_compile_options(-fno-semantic-interposition)
 endif()
+
 if (NOT WIN32)
     add_compile_options(-fPIC)
 endif()
+
+if (WIN32)
+    add_compile_options(-Wno-unknown-attributes) # [[no_unique_address]] is broken in MSVC ABI until next ABI break
+    add_compile_options(-Wno-expansion-to-defined) # FIXME: Complain to thakis about this one
```

Comment:
see patch descriptions of https://codereview.chromium.org/1584203002/ and https://reviews.llvm.org/D15866 for why this exists.

…why do you only need this on Windows though? Does it fire in system headers?

Reply:
Hm, this one never fires. Back in August when I first made these changes on a branch it was complaining loudly when using clang-cl.exe (I think?). I went and added the extra changes locally to let a clang-cl build work, but it never fires now 😕 . I'll get rid of it. Whatever it was that was causing this to happen has either been fixed in our code or in clang-cl since then.